containers/daap-create-schema/src/var/task/create_schema.py

```diff
@@ -0,0 +1,168 @@
+            keys_to_copy += [item["Key"] for item in page["Contents"]]
+    except KeyError as e:
+        logger.error(f"metadata for folder is empty but shouldn't be: {e}")
+    for key in keys_to_copy:
+        copy_source = {"Bucket": bucket, "Key": key}
+        destination_key = key.replace(latest_version, new_version)
+        s3_client.copy(
+            CopySource=copy_source,
+            Bucket=bucket,
+            Key=destination_key,
+            ExtraArgs=s3_security_opts,
+        )
+
+
+def handler(event, context):
+    data_product_name = event["pathParameters"].get("data-product-name")
+    table_name = event["pathParameters"].get("table-name")
+    logger = DataPlatformLogger(
+        extra={
+            "image_version": os.getenv("VERSION", "unknown"),
+            "base_image_version": os.getenv("BASE_VERSION", "unknown"),
+            "data_product_name": data_product_name,
+            "table_name": table_name,
+        }
+    )
+
+    logger.info(f"event: {event}")
+
+    request_body = json.loads(event.get("body"))
+
+    http_method = event.get("httpMethod")
+
+    if not http_method == HTTPMethod.POST:
+        error_message = f"Sorry, {http_method} isn't allowed."
+        logger.error(f"error message: {error_message}, input: {event}")
+        return response_status_400(error_message)
+
+    schema = DataProductSchema(
+        data_product_name=data_product_name,
+        table_name=table_name,
+        logger=logger,
+        input_data=request_body.get("schema"),
+    )
+
+    if schema.exists:
+        error_msg = (
+            f"v1 of this schema for table {table_name} already exists. You can upversion this schema if "
+            "there are changes from v1 using the PUT method of this endpoint. Or if this is a different "
+            "table then please choose a different name for it."
+        )
+        logger.error("create schema called where v1 already exists.")
+        return response_status_403(error_msg)
+
+    if not schema.has_registered_data_product:
+        error_msg = (
+            f"Schema for {table_name} has no registered metadata for the data product it belongs to. "
+            "Please first register the data product metadata using the POST method of the /data-product/register"
+            " endpoint."
+        )
+        logger.error("schema has no associated registered data product metadata.")
+        return response_status_403(error_msg)
+
+    # if schema already exist then we need to minor version increment to dataproduct metadata and schema
+    if schema.valid:
+        schema.convert_schema_to_glue_table_input_csv()
+        if not schema.parent_product_has_registered_schema:
+            metadata_dict = (
+                DataProductMetadata(
+                    data_product_name=data_product_name,
+                    logger=logger,
+                    input_data=None,
+                )
+                .load()
+                .latest_version_saved_data
+            )
```

Comment:
```suggestion
            metadata_dict = schema.parent_data_product_metadata
```
This looks to be reproducing [the logic from the DataProductSchema class](https://github.com/ministryofjustice/data-platform/blob/main/containers/daap-python-base/src/var/task/data_product_metadata.py#L223-L231)

Reply:
Yeah you're right, nice spot. Good to get a fresh pair of eyes on it